docs/proposals/kubelet-eviction.md

```diff
@@ -0,0 +1,201 @@
+# Kubelet - Eviction Policy
+
+**Author**: Derek Carr (@derekwaynecarr)
+
+**Status**: Proposed
+
+This document presents a specification for how the `kubelet` evicts pods when compute resources are too low.
+
+## Motivation
+
+The node needs a mechanism to preserve stability when available compute resources are low.
+
+## Scope of proposal
+
+This proposal defines a pod eviction policy for reclaiming compute resources.
+
+In the first iteration, it focuses on memory; later iterations are expected to cover
+other resources like disk.  The proposal focuses on a simple default core policy
+intended to cover the broadest class of user workloads.
+
+## Eviction Thresholds
+
+The `kubelet` will support the ability to define an eviction threshold when available
+compute resource drops below an administrator defined value.
+
+The `kubelet` will accept a flag to define the eviction threshold as follows:
+
+* `--eviction-threshold=:` - A set of `ResourceName`=`ResourceQuantity` pairs that
+describe per resource eviction thresholds.
+
+For example, an `--eviction-threshold=memory=256Mi` would mean that if the node has less than
+256Mi of available memory, the node may choose to initiate pod eviction.
+
+The node will only support `memory` first, but other resources such as `cpu` or `disk` may be
+added in the future.
+
+If no eviction threshold is specified for a compute resource, the node may choose a sane
```

Comment:
At the time of authoring this, I was trying to be consistent with what is in the Node Allocable proposal where it suggested a sane default.  I am fone defaulting to no-op if unspecified.